### PR TITLE
[SRM-948] Added d10 changes, cleaned up update hooks, updated deprecated functions.

### PR DESCRIPTION
**Pull request by MdNadimHossain**
**JIRA issue:** https://digital-vic.atlassian.net/browse/SRM-948

### Changed

1. Changes to make it compatible with D10.

### CODE REVIEW

File: modules/tide_share_link/src/Routing/Routes.php [hunk 1]
```diff
@@ -3,10 +3,10 @@
 namespace Drupal\tide_share_link\Routing;
 
 use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Routing\RouteObjectInterface;
```
Comment (MdNadimHossain) [inline modules/tide_share_link/src/Routing/Routes.php:6]: Regarding this change - https://www.drupal.org/node/3151009

File: modules/tide_share_link/src/EventSubscriber/ShareLinkTokenValidation.php [hunk 2]
```diff
@@ -10,7 +10,7 @@
 use Drupal\tide_share_link\PageCache\ShareLinkTokenPageCachePolicyInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
```
Comment (MdNadimHossain) [inline modules/tide_share_link/src/EventSubscriber/ShareLinkTokenValidation.php:13]: Update Symfony response and request events to new classes

File: src/EventSubscriber/TideApiVersionSubscriber.php [hunk 3]
```diff
@@ -3,7 +3,7 @@
 namespace Drupal\tide_api\EventSubscriber;
 
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
-use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
```
Comment (MdNadimHossain) [inline src/EventSubscriber/TideApiVersionSubscriber.php:6]: Update Symfony response and request events to new classes

**Review by mayurngondhkar — APPROVED** [review]
LGTM

**Review by anthony-malkoun — APPROVED** [review]
quick look and everything looks fine. Happy to be clearing out the old udpate hooks.

**Pull request merged**